client/service/src/builder.rs

```diff
@@ -53,8 +53,42 @@ use sysinfo::{get_current_pid, ProcessExt, System, SystemExt};
 use sc_telemetry::{telemetry, SUBSTRATE_INFO};
 use sp_transaction_pool::{TransactionPool, TransactionPoolMaintainer};
 use sp_blockchain;
-use grafana_data_source::{self, record_metrics};
+use prometheus_endpoint::{create_gauge, Gauge, U64, F64};
 
+prometheus_endpoint::lazy_static! {
+	pub static ref FINALITY_HEIGHT: Gauge<U64> = create_gauge(
+		"consensus_finality_block_height_number",
```

Comment:
These should be prefixed with something substrate specific. E.g. `substrate_consensus_finality[...]`. Or using `dots` like https://github.com/paritytech/dotexporter does.

Reply:
For more details on metric nameing, see https://prometheus.io/docs/practices/naming/.

Reply:
The best solution imo is to use a `Registry` with a custom prefix: https://docs.rs/prometheus/0.7.0/prometheus/struct.Registry.html#method.new_custom

Reply:
Since you can use the metric in polkadot later, **dot_** is better than **substrate_**.
try to prefix dot_ all Do you agree?

Reply:
The way I've got it set up in [my branch](https://github.com/paritytech/substrate/tree/ashley-prometheus) is that you can either pass in a Prometheus registry or have substrate create one for you, and this does the prefixing. This means that polkadot could pass in a different registry that prefixes all of substrate's metrics with `dot_` instead.